src/d3/hapticplot.d3.spec.ts

```diff
@@ -140,6 +137,33 @@ describe('VR Haptic Plot', () => {
     expect(getIntervalReset(controller)).toEqual(100);
   });
 
+  it('builds a k-d tree of the datapoints', () => {
+    const dataArray = [10, 15, 20, 25];
+    hapticplot.init(scene, dataArray);
+    expect(get3DTree(scene, shape)).toEqual({
+      position: new Vector3(0.35, 1.56, -0.35),
+      left: {
+        position: new Vector3(0.175, 1.42, -0.35),
+        left: {
+          position: new Vector3(0, 1.28, -0.35),
+          left: null,
+          right: null,
+        },
+        right: null,
+      },
+      right: {
+        position: new Vector3( 0.5249999999999999, 1.7, -0.35),
+        left: null,
+        right: null,
+      },
+    });
+  });
```

Comment:
I think when you move the KD tree into its own file, you may want a little more comprehensive testing, making sure that it works in some degenerative cases (like all the points are in a vertical line, or just a single point, or if the controller's position exactly equals one of the points in the data array.

This is good both as a general practice, but also in this particular case to make it easier to refactor the tree in case you want to optimize it in some way later on.  You can be more confident you didn't break anything with a bunch of tests.